packages/kit/src/runtime/server/page/load_node.js

```diff
@@ -51,19 +51,10 @@ export async function load_node({
 
 	let loaded;
 
-	const page_proxy = new Proxy(page, {
-		get: (target, prop, receiver) => {
-			if (prop === 'query' && prerender_enabled) {
-				throw new Error('Cannot access query on a page with prerendering enabled');
-			}
-			return Reflect.get(target, prop, receiver);
-		}
-	});
-
 	if (module.load) {
 		/** @type {import('types/page').LoadInput | import('types/page').ErrorLoadInput} */
 		const load_input = {
-			page: page_proxy,
+			page: create_page_proxy(page, prerender_enabled),
```

Comment:
let's change this — we don't want to incur the overhead of creating the proxy unnecessarily

```suggestion
			page: prerender_enabled ? create_page_proxy(page) : page,
```